### PR TITLE
Introducing Alive & Ready endpoint

### LINKED ISSUE #28: Live and Ready endpoint support
## Is your feature request related to a problem? Please describe

We have to introduce readiness and liveness endpoint to support checks from kubernetes and other systems.

The liveness indicates that the service is running correctly. Any code greater than or equal to 200 and less than 400 indicates success. Any other code indicates failure.
The endpoint will, by default return 200 and will let the developer change this to 500 on demand.
The actual endpoint will be **/alive**.

The readiness indicates that the service can accept incoming traffic. The endpoint will, by default return 200 and will let the developer change in order to transition from 500 to 200.
The endpoint for readiness will be **/ready**.

The existing /health endpoint will be removed.

## Additional context

There is already a issue https://github.com/thebeatapp/patron/issues/24 which tries to introduce the healthiness concept in components. If the above works as design we should not implement the change in the component since it will introduce a lot of unneeded complexity.

### LINKED ISSUE #92: Introduce /health and /metrics sync_http_requests prometheus metrics
## Is your feature request related to a problem? Please describe

All HTTP routes are measured have prometheus metrics except /health and /metrics and they too are technically HTTP routes

## Is this a bug? Please provide steps to reproduce, a failing test etc.

Not a bug

## Describe the solution

Monitor the `/health` and `/metrics` routes the same way Patron tracks other routes

### PR DESCRIPTION
**Pull request by asenlog**
## Which problem is this PR solving?

This PR is proposed to resolve #28 

## Short description of the changes

This PR introduces /alive and /ready endpoint and removes /health

### CODE REVIEW

File: sync/http/alive.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import (
+	"net/http"
+)
+
+// AliveStatus type representing the liveness of the service via HTTP component.
+type AliveStatus int
+
+const (
+	// Initializing represents a state when warming up and before the component is Alive.
+	Initializing AliveStatus = 0
```
Comment (mantzas) [inline sync/http/alive.go]: The liveness has 2 states: alive (HTTP 200 and not HTTP 500). We do not need a third state anymore.

**Review by mantzas — CHANGES_REQUESTED** [review]
Hi, some things are not done yet:

- The liveness is not exposed via HTTP.
- The old /health route has not been removed.
- The readiness implementation is missing.

File: sync/http/ready.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+type ReadyCheckFunc func() ReadyStatus
+
+func readyCheckRoute(rcf ReadyCheckFunc) Route {
+
+	f := func(w http.ResponseWriter, r *http.Request) {
+		switch rcf() {
+		case Ready:
+			w.WriteHeader(http.StatusOK)
+		case NotReady:
+			w.WriteHeader(http.StatusInternalServerError)
```
Comment (mantzas) [inline sync/http/ready.go]: I would use: 503 Service Unavailable

File: sync/http/alive.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+type AliveCheckFunc func() AliveStatus
+
+func aliveCheckRoute(acf AliveCheckFunc) Route {
+
+	f := func(w http.ResponseWriter, r *http.Request) {
+		switch acf() {
+		case Alive:
+			w.WriteHeader(http.StatusOK)
+		case Unresponsive:
+			w.WriteHeader(http.StatusInternalServerError)
```
Comment (mantzas) [inline sync/http/alive.go]: Maybe 503 Service Unavailable?

**Review by mantzas — CHANGES_REQUESTED** [review]

**Review by mantzas — DISMISSED** [review]

**Review by sotirispl — DISMISSED** [review]

File: sync/http/ready_test.go [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_readyCheckRoute(t *testing.T) {
```
Comment (cpapidas) [inline sync/http/ready_test.go:11]: could we rename it to `TestReadyCheckRoute` for consistency?

File: sync/http/alive_test.go [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_aliveCheckRoute(t *testing.T) {
```
Comment (cpapidas) [inline sync/http/alive_test.go:11]: Could we rename this function to `TestAliveCheckRoute ` for consistency?

**Review by cpapidas — DISMISSED** [review]

File: sync/http/ready_test.go (hunk 4 above)
Comment (asenlog) [reply sync/http/ready_test.go:11]: Kept it that ways as there are already other tests following the same convention. You can have a look ok component_test, profiling_test, metrics_test and also the old health_test was following that same convention.

File: sync/http/alive_test.go (hunk 5 above)
Comment (asenlog) [reply sync/http/alive_test.go:11]: Kept it that ways as there are already other tests following the same convention. You can have a look ok component_test, profiling_test, metrics_test and also the old health_test was following that same convention.

**Review by mantzas — APPROVED** [review]

**Review by mantzas — APPROVED** [review]

**Review by sotirispl — APPROVED** [review]

**Review by cpapidas — APPROVED** [review]

**Pull request merged**